### PR TITLE
Bimbofication part 1 (orgasms sorted)

### PR DESCRIPTION
**Pull request by losdivo**
First part of bimbofication code.  In this part, all orgasms are redirected to a wrapper functions, and statistics on types of orgasms are collected.

The full description of bimbofication is available here:

https://docs.google.com/document/d/1MplwdEk5ilEEgF66aaUDz-EXfFiuxcPNYXZZqsOZCqo/edit

### CODE REVIEW

**Comment by Stadler76** [comment]
mmh, still 170 changed files ...

Why don't you do something, like the following:
1. Rename (not refactor!) the method `orgasm()` to `orgasmReal()`
2. create a the new method `orgasm()` in Creature.as:
```as3
public function orgasm(type:String = 'Default', real:Boolean = true):void
{
	switch (type) {
		// Start with that, whats easy
		case 'Vaginal': flags[kFLAGS.TIMES_ORGASM_VAGINAL]++; break;
		case 'Anal':    flags[kFLAGS.TIMES_ORGASM_ANAL]++;    break;
		case 'Dick':    flags[kFLAGS.TIMES_ORGASM_DICK]++;    break;
		case 'Lips':    flags[kFLAGS.TIMES_ORGASM_LIPS]++;    break;
		case 'Tits':    flags[kFLAGS.TIMES_ORGASM_TITS]++;    break;
		case 'Nipples': flags[kFLAGS.TIMES_ORGASM_NIPPLES]++; break;
		case 'Ovi':     break;

		// No to the more complex types
		case 'VaginalAnal':
			orgasm((hasVagina() ? 'Vaginal' : 'Anal'), real);
			return; // Prevent calling orgasmReal() twice

		case 'DickAnal':
			orgasm((rand(2) == 0 ? 'Dick' : 'Anal'), real);
			return;

		case 'Default':
		case 'Generic':
		default:
			if (hasVagina() && hasCock()) {
				orgasm((rand(2) == 0 ? 'Vaginal' : 'Dick'), real);
				return;
			}

			orgasm((hasVagina() ? 'Vaginal' : 'Dick'), real);
			return;
	}

	if (real) orgasmReal();
}
```

3. Leave all calls to `player.orgasm();` untouched for now (or revert them in your branch)
4. File the PR

For now, all orgasms will be default orgasms, but the benefit would be, that is affects far less, than 170 files.

In later PRs you could change that in batches of files (e. g. 10-20 files each PR).
There you change `player.orgasm()` to:
- `player.orgasm('Vaginal');`
- `player.orgasm('Anal');`
- `player.orgasm('VaginalAnal');`
- `player.orgasm('Dick');`
- and so on ...

And leave it as is, where you choose the default orgasm, since `player.orgasm('Default')` and `player.orgasm('Generic')` would be the same, as `player.orgasm()`

Unless you choose something else to be the default, that is.

Well, I hope, you get the point.

**Comment by Stadler76** [comment]
btw: I've chosen to use the ternary operator:
```as3
orgasm((rand(2) == 0 ? 'Vaginal' : 'Dick'), real);
```
instead of if-else:
```as3
if (rand(2) == 0)
	orgasm('Vaginal', real);
else
	orgasm('Dick', real);
```
in order to ave a few lines of code. I hope, its not too cluttered for ya.

File: classes/classes/GlobalFlags/kFLAGS.as [hunk 1]
```diff
@@ -2905,13 +2905,13 @@ public static const UNKNOWN_FLAG_NUMBER_02896:int
...
-public static const UNKNOWN_FLAG_NUMBER_02904:int                                   = 2904;
-public static const UNKNOWN_FLAG_NUMBER_02905:int                                   = 2905;
-public static const UNKNOWN_FLAG_NUMBER_02906:int                                   = 2906;
+public static const TIMES_ORGASM_VAGINAL:int 	 									= 2900; // 
+public static const TIMES_ORGASM_ANAL:int 											= 2901;
+public static const TIMES_ORGASM_DICK:int											= 2902;
+public static const TIMES_ORGASM_TITS:int                                           = 2903;
+public static const TIMES_ORGASM_NIPPLES:int                                        = 2904;
+public static const TIMES_ORGASM_LIPS:int                                        	= 2905;
+public static const BIMBOSKIRT_MINIMUM_LUST:int          							= 2906;
```
Comment (Stadler76) [inline classes/classes/GlobalFlags/kFLAGS.as]: You should use spaces for alignment there, not tabs.

**Comment by losdivo** [comment]
Ok, sounds reasonable. There will no way to catch new default orgasms, but the number of changes will be much smaller.

**Comment by Stadler76** [comment]
Sure. And like I said: Leave the calls to `player.orgasm();` as is and wade through the code to change them in later commits and/or PRs.

**Comment by losdivo** [comment]
Here you go. Only 2 files changed.

**Comment by Stadler76** [comment]
Just checked out your branch and grepped through the files to count the calls to your orgasmWhatever-methods. Heres the statistic:

| call                         | count |
|------------------------------|-------|
| `player.orgasmAnal()`        | 115   |
| `player.orgasmVaginal()`     | 285   |
| `player.orgasmDick()`        | 570   |
| `player.orgasmLips()`        | 29    |
| `player.orgasmTits()`        | 44    |
| `player.orgasmNipples()`     | 4     |
| `player.orgasmOvi()`         | 27    |
| `player.orgasmVaginalAnal()` | 28    |
| `player.orgasmDickAnal()`    | 1     |
| `player.orgasmGeneric()`     | 383   |

mmh, orgasmDick wins, but I guess, using orgasmGeneric makes more sense to be the default.
Well, you decide.

**Comment by losdivo** [comment]
I put `player.orgasmDick()` in the scenes where player is fucking someone with a dick, or player's dick is being sucked. I guess there are a lot more such scenes than generic sexual scenes where the source of pleasure is not specified.

**Pull request merged**